pyrk/th_component.py

```diff
@@ -83,33 +126,164 @@ def update_temp(self, timestep, temp):
         :param timestep: the timestep at which to query the temperature
         :type timestep: int
         :param temp: the new tempterature
-        :type float: float, units of kelvin
+        :type float: float, dimensionless
         """
         self.T[timestep] = temp
         self.prev_t_idx = timestep
         return self.T[timestep]
 
-    def dtemp(self):
+    def dtemp(self, timestep, T0_timestep):
         if self.prev_t_idx == 0:
             return 0.0*units.kelvin
         else:
-            return (self.T[self.prev_t_idx] - self.T[self.prev_t_idx-1])
+            T0 = self.T[T0_timestep]
+            return self.T[timestep-1]-T0
+        # return (self.T[self.prev_t_idx] - self.T[self.prev_t_idx-1])
 
-    def temp_reactivity(self):
-        return self.alpha_temp*self.dtemp()
+    def temp_reactivity(self, timestep, T0_timestep):
+        '''alpha_temp is converted to deltak/kelvin'''
+        return self.alpha_temp*self.dtemp(timestep, T0_timestep)
 
     def add_convection(self, env, h, area):
         self.conv[env] = {
             "h": h.to('joule/second/kelvin/meter**2'),
-            "area": area.to('meter**2')
+            "area": area
         }
 
-    def add_conduction(self, env, area):
-        self.cond[env] = area.to('meter**2')
-
     def add_mass_trans(self, env, H, u):
         self.mass[env] = {"H": H,
                           "u": u}
 
     def add_custom(self, env, res):
         self.cust[env] = {"res": res.to(units.kelvin/units.watt)}
+
+    def addConvBC(self, env, prev_comp, h, R):
+        self.convBC[env] = {
+            "h": h.to('joule/second/kelvin/meter**2'),
+            "prev_comp": prev_comp,
+            "R": R
+        }
+
+    def add_conduction(self, env, k,
+                       area=0.0*units.meter**2, L=0.0*units.meter,
+                       r_b=0.0*units.meter, r_env=0.0*units.meter):
+        '''Add parameters for conduction heat transfer calculation
+        area and L are used for slab geometry
+        r_b and r_env are used for spherical heat diffusion
+        Args:
```

Comment:
As noted in the contributor guidelines, we use the sphinx autodoc syntax to build our online automated documentation. This "Args" syntax is a bit easier to read, but doesn't give us the pretty spinx formatting we're looking for in the automated documentation.

https://pythonhosted.org/an_example_pypi_project/sphinx.html#function-definitions